server/parsec/components/memory/shamir.py

```diff
@@ -112,7 +119,23 @@ async def add_recovery_setup(
 
         if self._data.organizations[organization_id].shamir_setup.get(author) is None:
             self._data.organizations[organization_id].shamir_setup[author] = MemoryShamirSetup(
-                setup.ciphered_data, setup.reveal_token, brief, shares
+                setup.ciphered_data, setup.reveal_token, brief, shares, setup.brief
             )
         else:
             return ShamirAddOrDeleteRecoverySetupStoreBadOutcome.ALREADY_SET
+
+    @override
+    async def get_brief_certificate(
```

Comment:
Should we check that the user exists in the organization, and then return a specific error ?